packages/react-select/src/accessibility/index.js

```diff
@@ -1,92 +1,136 @@
+import {
+  type ActionMeta,
+  type OptionType,
+  type OptionsType,
+  type ValueType,
+} from '../types';
 
-export type InstructionsContext = {
+export type FocusedType = 'option' | 'value';
+
+export type GuidanceType = 'menu' | 'input' | 'value';
+
+export type AriaLiveProp = 'polite' | 'off' | 'assertive';
+
+export type OptionContext = {
+  // derived label of selected option via getOptionLabel
+  // Note: different aria attributes could potentially introduce need for DOM query selectors
+  label?: string,
+  // selected option was disabled, used only for accessibility purposes
+  isDisabled?: boolean,
+  // option is selected
+  isSelected?: boolean,
+};
+
+export type SelectionContext = ActionMeta &
+  OptionContext & {
+    // selected "thing" (option, removedValue, value)
+    selected?: OptionType | null,
+  };
+
+export type FocusedContext = OptionContext & {
+  type: FocusedType,
+  options: OptionsType,
+  value?: ValueType,
+};
+
+export type GuidanceContext = OptionContext & {
   isSearchable?: boolean,
   isMulti?: boolean,
-  label?: string,
   isDisabled?: boolean,
   tabSelectsValue?: boolean,
 };
-export type ValueEventContext = { value: string, isDisabled?: boolean };
-
-export const instructionsAriaMessage = (
-  event: string,
-  context?: InstructionsContext = {}
-) => {
-  const { isSearchable, isMulti, label, isDisabled, tabSelectsValue } = context;
-  switch (event) {
-    case 'menu':
-      return `Use Up and Down to choose options${
-        isDisabled ? '' : ', press Enter to select the currently focused option'
-      }, press Escape to exit the menu${
-        tabSelectsValue
-          ? ', press Tab to select the option and exit the menu'
-          : ''
-      }.`;
-    case 'input':
-      return `${label ? label : 'Select'} is focused ${
-        isSearchable ? ',type to refine list' : ''
-      }, press Down to open the menu, ${
-        isMulti ? ' press left to focus selected values' : ''
-      }`;
-    case 'value':
-      return 'Use left and right to toggle between focused values, press Backspace to remove the currently focused value';
-  }
+
+export type AriaSelectionType = SelectionContext & {
+  // first parameter passed in onChange
+  value?: ValueType,
 };
 
-export const valueEventAriaMessage = (
-  event: string,
-  context: ValueEventContext
-) => {
-  const { value, isDisabled } = context;
-  if (!value) return;
-  switch (event) {
-    case 'deselect-option':
-    case 'pop-value':
-    case 'remove-value':
-      return `option ${value}, deselected.`;
-    case 'select-option':
-      return isDisabled
-        ? `option ${value} is disabled. Select another option.`
-        : `option ${value}, selected.`;
-  }
+export type AriaLiveMessagesProps = {
+  onChange?: (value: ValueType, context: SelectionContext) => string,
+  onFocus?: (focused: OptionType, context: FocusedContext) => string,
+  onFilter?: (args: {
+    inputValue: string,
+    screenReaderMessage: string,
```

Comment:
`screenReaderMessage` should be `resultsMessage`

Reply:
Replaced this with single prop so should be resolved now.